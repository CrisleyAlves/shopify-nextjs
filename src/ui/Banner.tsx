import Image from "next/image";
import Link from "next/link";
import React from "react";

const Banner = (): React.ReactElement => {
  return (
    <Link href="/collections/best-sales" className="w-full">
      <Image
        priority
        src="/banner2.jpg"
        alt="Product"
        width={0}
        height={0}
        sizes="100vw"
        className="w-full h-auto lg:h-[500px]"
      />
    </Link>
  );
};

export default Banner;
