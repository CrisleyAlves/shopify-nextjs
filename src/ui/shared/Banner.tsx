import Image from "next/image";
import Link from "next/link";

const Banner = (): React.ReactElement => {
  return (
    <Link href="/collections/sales" className="w-full">
      <Image
        priority
        src="/banner.webp"
        alt="Product"
        width={0}
        height={0}
        sizes="100vw"
        className="w-full h-auto lg:h-[500px]"
      />
    </Link>
  );
};

export default Banner;
