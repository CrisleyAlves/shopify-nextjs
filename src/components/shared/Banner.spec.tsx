import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";

import Banner from "./Banner";

describe("Banner", () => {
  it("renders a link to /collections/sales", () => {
    const { container } = render(<Banner />);
    const link = screen.getByRole("link");

    expect(link).toHaveAttribute("href", "/collections/sales");
    expect(container).toMatchSnapshot();
  });
});
