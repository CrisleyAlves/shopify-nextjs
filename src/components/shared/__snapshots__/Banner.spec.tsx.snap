// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Banner renders a link to /collections/sales 1`] = `
<div>
  <a
    class="w-full"
    href="/collections/sales"
  >
    <img
      alt="Product"
      class="w-full h-auto lg:h-[500px]"
      data-nimg="1"
      decoding="async"
      height="0"
      sizes="100vw"
      src="/_next/image?url=%2Fbanner.webp&w=3840&q=75"
      srcset="/_next/image?url=%2Fbanner.webp&w=640&q=75 640w, /_next/image?url=%2Fbanner.webp&w=750&q=75 750w, /_next/image?url=%2Fbanner.webp&w=828&q=75 828w, /_next/image?url=%2Fbanner.webp&w=1080&q=75 1080w, /_next/image?url=%2Fbanner.webp&w=1200&q=75 1200w, /_next/image?url=%2Fbanner.webp&w=1920&q=75 1920w, /_next/image?url=%2Fbanner.webp&w=2048&q=75 2048w, /_next/image?url=%2Fbanner.webp&w=3840&q=75 3840w"
      style="color: transparent;"
      width="0"
    />
  </a>
</div>
`;
